### ISSUE TITLE
Default pin goes to temporary directory?

### ISSUE THREAD

**Issue by hadley** [rstudio/pins#39]
``` r
pins::pin("https://raw.githubusercontent.com/facebook/prophet/master/examples/example_retail_sales.csv")
#> [1] "/private/tmp/RtmpK3Ni0B/file8ebc1adfe617/temp/example_retail_sales/example_retail_sales.csv"
```

I found this surprising, since I thought the point of `pin()` was to cache it persistently

**Comment by jjallaire** [comment]
CRAN complained about writing to the user's home directory without their permission (and pointed to the exact line of code which was doing this). So the behavior was changed to only write persistently if the user pins to their local board.

My view is that using `rappdirs` to write to the user's cache directory should be fine and that requiring explicit use of the local board will rob many users of the benefits of caching. I wonder if the verb 'pin' essentially implies persistent local caching so that really constitutes "permission" (i.e. if the package is advertised as providing a local cache and you use the package that seems like permission to me).

**Comment by kevinushey** [comment]
FWIW I'm facing similar challenges with `renv`. The approach I'm taking right now is:

1. Show a one-time startup banner in which we tell the user about `renv` and the directory we use for caching;

2. If the user is okay with that, then we create the directory and use it from then on;

3. Otherwise, cached items always enter the temporary directory.

Users can also "implicitly" consent by setting e.g. the `RENV_PATHS_ROOT` environment variable to point to their own desired cache root.

**Comment by jjallaire** [comment]
I like the idea of interactively asking for consent (as opposed to users
having to learn about and use board_register_local)

**Comment by hadley** [comment]
From what I hear, CRAN is generally ok with the use of rappdirs.

**Comment by hadley** [comment]
The problem (I think) with interactively asking for consent, is that you have to record if they decline, which basically means you need to solve the problem of where to write on disk anyway.

**Comment by jjallaire** [comment]
Yeah my feeling was that they are fine with rappdirs (and the large
number of dependencies on rappdirs seems to imply this is true).
Javier, what would you think about just using the OS-defined user data
cache directory from rappdirs without asking for consent?

**Comment by javierluraschi** [comment]
Yeah, this probably deserves more thought in a subsequent release. One idea I have is adding support for the config package, and add a board_persist() function which would persist your registered boards to the config file, potentially a parent folder that all projects can share. When the config file is found, pins would register all the boards automatically.

**Comment by jjallaire** [comment]
I think it's important that users get the caching automatically w/o any
manual steps or reading of the docs. Users that copy/paste code that uses
pins to get access to datasets won't generally bother reading all about pin
persistence, they'll just get the default temp directory behavior without
further thought (i.e. many 'users' of pins won't even know it's there and
definitely won't read the docs)

**Comment by javierluraschi** [comment]
One option would be to add in RStudio the option:

```
options(pins.board = "local")
```

Which would switch the default board to be a local board with a `~/.pins` cache, but would prefer to avoid different behavior.

I could also check in the package and change the default for RStudio sessions.

This behavior could be clearly explained in the RStudio vignette.... thoughts?

**Comment by javierluraschi** [comment]
And yes, I'll try first to use the `OS-defined user data cache directory` with `rappdirs`. Next major release I'll give it a shot and report back, if that fails, maybe we can do the config or RStudio workaround.

**Comment by jjallaire** [comment]
Yes, I think not using a local cache by default leaks a huge part of the
benefits of the package for more casual users.

**Comment by Robinlovelace** [comment]
Thinking about using this package for caching data in other packages, do you think it would be suitable for that? I noticed that it was silently using `tempdir()` for pinning, which of course is non-persistent. I think the documentation could be more explicit about that, was only after some playing and then discovering this issue that I discovered that. I agree that asking users to set

```
options(pins.board = "local")
```

would be a good solution but would that persist? I imagine it would need to go in the .Rprofile. 

Thanks for a great package, for what it's worth I'm considering using it in this prototype package for accessing OSM data: https://github.com/ITSLeeds/geofabric/

**Comment by javierluraschi** [comment]
This one was fixed in `pins 0.2.0`.

@Robinlovelace the package now defaults to `local` so no need to run `options(pins.board = "local")` and yes, I think is suitable for caching files, see for instance https://github.com/r-tensorflow/gpt2/blob/master/R/download.R#L21. Otherwise, each package caches data in different location and users don't know what is really being cached, with pins you get to see what is being cached and easily remove that from your local environement.

**Comment by Robinlovelace** [comment]
Thanks for the update @javierluraschi
